query-engine/prisma-models/src/model.rs

```diff
@@ -1,61 +1,55 @@
 use crate::prelude::*;
-use once_cell::sync::OnceCell;
-use psl::schema_ast::ast;
-use std::{
-    fmt::Debug,
-    hash::{Hash, Hasher},
-    sync::{Arc, Weak},
-};
+use psl::{parser_database::walkers, schema_ast::ast};
 
-pub type ModelRef = Arc<Model>;
-pub type ModelWeakRef = Weak<Model>;
+pub type Model = crate::Zipper<ast::ModelId>;
+pub type ModelRef = Model;
+pub type ModelWeakRef = Model;
 
-pub struct Model {
-    pub id: ast::ModelId,
-    pub(crate) name: String,
-    pub(crate) manifestation: Option<String>,
-    pub(crate) fields: OnceCell<Fields>,
-    pub(crate) indexes: OnceCell<Vec<Index>>,
-    pub(crate) primary_identifier: OnceCell<FieldSelection>,
-    pub(crate) dml_model: dml::Model,
+// pub struct Model {
```

Comment:
que?

Reply:
good catch, more code we can delete :fire: